src/main/kotlin/nl/nuts/discovery/store/entity/NutsCertificateRequest.kt

```diff
@@ -0,0 +1,159 @@
+/*
+ *     Nuts discovery service
+ *     Copyright (C) 2020 Nuts community
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package nl.nuts.discovery.store.entity
+
+import net.corda.core.identity.CordaX500Name
+import org.bouncycastle.asn1.ASN1ObjectIdentifier
+import org.bouncycastle.asn1.DERTaggedObject
+import org.bouncycastle.asn1.DERUTF8String
+import org.bouncycastle.asn1.DLSequence
+import org.bouncycastle.asn1.pkcs.Attribute
+import org.bouncycastle.asn1.pkcs.PKCSObjectIdentifiers
+import org.bouncycastle.asn1.x509.Extension
+import org.bouncycastle.asn1.x509.Extensions
+import org.bouncycastle.asn1.x509.GeneralName
+import org.bouncycastle.asn1.x509.GeneralNames
+import org.bouncycastle.openssl.PEMParser
+import org.bouncycastle.pkcs.PKCS10CertificationRequest
+import java.io.BufferedReader
+import java.io.ByteArrayInputStream
+import java.io.InputStreamReader
+import java.io.StringReader
+import java.time.LocalDateTime
+import javax.persistence.Column
+import javax.persistence.Entity
+import javax.persistence.GeneratedValue
+import javax.persistence.GenerationType
+import javax.persistence.Id
+
+
+/**
+ * Entity to store PEM encoded CSR in DB
+ */
+@Entity
+class NutsCertificateRequest {
+
+    companion object {
+        val NUTS_VENDOR_EXTENSION: ASN1ObjectIdentifier = ASN1ObjectIdentifier("1.3.6.1.4.1.54851.4").intern()
+
+        /**
+         * Create entity from PKCS10CertificationRequest, stores .encoded as bytes
+         */
+        fun fromPEM(pem: String): NutsCertificateRequest {
+            val req = NutsCertificateRequest().apply {
+                val csr = parsePEM(pem)
+
+                name = CordaX500Name.parse(csr.subject.toString()).toString() // this puts stuff in right order
+                this.pem = pem
+                submittedAt = LocalDateTime.now()
+                oid = extractOID(csr)
+            }
+
+            if (req.oid == null) {
+                throw IllegalArgumentException("Given CSR does not have a correctly formatted oid in extensions")
+            }
+
+            return req
+        }
+
+        /**
+         * Parse a pem-encoded CSR
+         */
+        fun parsePEM(pem: String): PKCS10CertificationRequest {
```

Comment:
parsePEM vs fromPEM with different implementation and return types deserve more explicit naming.

Reply:
changed name of this one to `pemToPKCS10`